lib/snippets/publish-lerna-independent-packages

```diff
@@ -6,31 +6,30 @@ const taggedPackages = execSync("git tag --points-at HEAD")
   .trim()
   .split("\n");
 
-const hasBetaRelease = taggedPackages.some((version) =>
-  version.includes("-beta")
-);
+// anything that matches `-` after MAJOR.MINOR.PATCH
+const validPattern = new RegExp(/\d+\.\d+\.\d+-(\d+|\w+)/);
+
+// will be null if a full release
+const baseMatch = validPattern.exec(taggedPackages[0]);
+
+const isValid = taggedPackages.every((version) => {
+  const match = validPattern.exec(version);
+  if (baseMatch === null) {
+    return baseMatch === match;
+  }
+  return baseMatch[1] === match[1];
+});
 
 // set this using machine.environment.SHIPIT_LERNA_PUBLISH_FROM_GIT in your shipit.yml
 const useFromGit = process.env.SHIPIT_LERNA_PUBLISH_FROM_GIT || false;
```

Comment:
Not related to this PR but in retrospect perhaps this should have been `SHIPIT_LERNA_PUBLISH_MECHANISM` and have it default to `from-package`, and users could set `SHIPIT_LERNA_PUBLISH_MECHANISM='from-git'`, to allow for possible new approaches in the future.


I'm also not that convinced about from-package being the default - it feels like `from-git` would be preferable if we're doing stuff like asserting that this commit has a bunch of tags on it. `from-package` doesn't use any of that info